src/main/java/com/maddyhome/idea/vim/extension/surround/VimSurroundExtension.kt

```diff
@@ -181,21 +181,25 @@ internal class VimSurroundExtension : VimExtension {
           }
 
           val registerValue = getRegisterForCaret(REGISTER, it.caret)
-          val innerValue = if (registerValue.isNullOrEmpty()) null else registerValue
+          val innerValue = if (registerValue.isNullOrEmpty()) emptyList() else registerValue
           it.innerText = innerValue
-        }
 
-        surroundings.forEach {
-          if (it.innerText == null && getRegisterForCaret(REGISTER, it.caret)?.isNotEmpty() == true) {
-            it.innerText = emptyList()
```

Comment:
This condition won't be ever `true`, because of the lines above (L183-L185). 

Those lines set `innerText` to `null` if register is null or empty. But here we loop through surroundings again and try to find `innerText == null` for cases when register is not empty.